core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -205,6 +205,7 @@ private static void validateAndAddTemplate(final PutRequest request, IndexTempla
 
             //create index service for parsing and validating "mappings"
             Settings dummySettings = Settings.builder()
+                .put(request.settings)
```

Comment:
Two lines above there is a decision made about the dummy number of shards made but not all of the settings were used previously.